internal/plans/internal/planproto/planfile.proto

```diff
@@ -126,6 +129,7 @@ enum ResourceInstanceActionReason {
     REPLACE_BECAUSE_TAINTED = 1;
     REPLACE_BY_REQUEST = 2;
     REPLACE_BECAUSE_CANNOT_UPDATE = 3;
+    DRIFT_DETECTED = 4;
```

Comment:
This overall approach makes sense to me, but I feel a little unsure about this "action reason" since it seems to be reporting something a bit different than what the other "reasons" are talking about: we're not saying "we're updating this because we detected drift", but rather "this change object is describing drift rather than a planned action". (At least, that's what I understood from the rest of this... :thinking: )

In particular, it seems like this design would make it hard to hypothetically report a reason _for_ the drift, if we needed to do such a thing in the future. (I don't know what that would be, so perhaps this is a non-issue.)

With that said then, I guess my intuition here was that the placement of a change object inside `Plan.DriftChanges` vs. `Plan.Changes` is already enough of a signal that it's reporting "drift" instead of a planned action. Do we actually need to additionally annotate the individual changes? Is there a situation where we might expect to find `DRIFT_DETECTED` changes mixed in other reasons in either of the "changes" fields?

Reply:
I see what you mean here. Embedding some kind of marker in the `ResourceInstanceChangeSrc` value itself allows `format.ResourceChange` to render the different summary text we currently have for drifted vs planned changes. Without that information in the change object, I think we'd have to add a (boolean?) argument to the function to achieve the same job. If that makes more sense to you I can certainly do that.

Reply:
I think making the UI layer deal with this concern of using different language for each of them does make more sense to me. 

Perhaps instead of a boolean it could be an enum initially with just two members, so that we can give them names that will hopefully make the callsites easier to read: a `type DiffLanguage` with members `DetectedDriftLanguage` and `ProposedChangeLanguage`, perhaps? (This is just an idea; I don't feel strongly about it.)